jina/hubble/helper.py

```diff
@@ -2,13 +2,65 @@
 
 import io
 import os
+import hashlib
+from typing import Tuple
 import zipfile
 from pathlib import Path
-
+from urllib.parse import urlparse
 from jina import __resources_path__
 from jina.importer import ImportExtensions
 
 
+def parse_hub_uri(uri_path: str) -> Tuple[str, str, str, str]:
+    """Parse the uri of the Jina Hub executor.
+
+    :param uri_path: the uri of Jina Hub executor
+    :return: a tuple of schema, id, tag, and secret
+    """
+    parser = urlparse(uri_path)
+    scheme = parser.scheme
+    items = list(parser.netloc.split(':'))
+    id = items[0]
```

Comment:
what if list is empty?

Reply:
Good question. I will check the `uri_path`, and throw an exception if necessary.